### PR TITLE
Create TrendingNftCollections and type some methods

### PR DESCRIPTION
**Pull request by emanuelpinho**
This PR aims to create the new TrendingNftCollections. 

Since this method uses the conversion helper `removeNodesAndEdges.ts`, there was a need to create new typings that could support `codegen`. Because of that, it was added some typing files and hence the size of this pull request, sorry about this, though it's mainly recordings and typings.

### CODE REVIEW

File: packages/libs/api/sdk/codegen.yml [hunk 1]
```diff
@@ -6,4 +6,7 @@ generates:
     plugins:
       - typescript
       - typescript-operations
-      - typed-document-node
+      - typescript-document-nodes
+  src/graphql/fragmentMatcher.ts:
```
Comment (emanuelpinho) [inline packages/libs/api/sdk/codegen.yml:10]: I think in the near future we want to standardize the return types. I added this codegen plugin so we can make sure we keep consistency between multiple requests

**Review by AlejandroFabianCampos — COMMENTED** [review]
Awesome stuff Emanuel!
Just noticed that after building, some types work and some are referring to a file that is not being bundled seemingly 
(getTrendingNFTCollections is typed correctly)
image
(getNFTsByWalletENS is not picking up the type correctly)
image
image
these types should come from ./sharedTypes from that declaration file but that's not on the bundle

**Comment by emanuelpinho** [comment]
> Awesome stuff Emanuel! Just noticed that after building, some types work and some are referring to a file that is not being bundled seemingly (getTrendingNFTCollections is typed correctly) image (getNFTsByWalletENS is not picking up the type correctly) image image these types should come from ./sharedTypes from that declaration file but that's not on the bundle

Ok, I managed to change some stuff. 
- Converted that to codegen as well
- Changes from d.ts to .ts file. We were exporting some interfaces in that file, which I think it's not intended for a global d.ts file. If we want to have a global type file so we should include it in the tsconfig and make use of it in the client, which I don't think it's going to be the case

**Comment by emanuelpinho** [comment]
> Awesome stuff Emanuel! Just noticed that after building, some types work and some are referring to a file that is not being bundled seemingly (getTrendingNFTCollections is typed correctly) image (getNFTsByWalletENS is not picking up the type correctly) image image these types should come from ./sharedTypes from that declaration file but that's not on the bundle

Ok, I managed to change some stuff. 
- Converted that to codegen as well
- Changes from d.ts to .ts file. We were exporting some interfaces in that file, which I think it's not intended for a global d.ts file. If we want to have a global type file so we should include it in the tsconfig and make use of it in the client, which I don't think it's going to be the case

**Review by johnpmitsch — COMMENTED** [review]
Love these changes! I think this will help our typing quite a bit

A couple suggestions and questions:
- For `getTrendingNFTCollections`, can you add it to the docs?
- For the codegen, this is really cool, I'm still ramping up on it so forgive me if this is a dumb question, but how would I know the possible values for the inputs here from the documentation? For example,  the `timePeriod` input takes `ONE_HOUR`, `TWELVE_HOURS`, `ONE_DAY`, `SEVEN_DAYS` but it shows `InputMaybe<TrendingCollectionsTimePeriodEnum>`, which I would have to then look up? Is it able to be more specific?
Screen Shot 2022-11-04 at 10 52 29 AM

File: packages/libs/api/sdk/src/queries/nft/getTrendingNFTCollections/getTrendingNFTCollections.gql [hunk 2]
```diff
@@ -0,0 +1,30 @@
+query TrendingNFTCollections(
+  $after: String
+  $first: Int
+  $orderBy: TrendingCollectionsOrderByEnum
+  $orderDirection: OrderDirectionEnum
+  $timePeriod: TrendingCollectionsTimePeriodEnum
+  $timeRange: DateInputType
+  $marketplace: OrderMarketplaceInput
```
[the remark this answers is not in the document]
Comment (johnpmitsch) [reply packages/libs/api/sdk/src/queries/nft/getTrendingNFTCollections/getTrendingNFTCollections.gql:8]: I think this is being done with codegen, so I'm going to resolve this, I didn't understand it at the time! 😆

File: packages/libs/api/sdk/src/queries/fragments/collection.gql [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    name
+    symbol
+    stats {
+      average
+      ceiling
+      floor
+      totalSales
+      volume
+    }
+    tokens(first: 5) {
```
[the remark this answers is not in the document]
Comment (emanuelpinho) [reply packages/libs/api/sdk/src/queries/fragments/collection.gql:15]: Oh, perfect thing. I didn't know about this page. Sure it makes sense to me to adapt to this

**Review by AlejandroFabianCampos — COMMENTED** [review]
Typing resolution definitely improved! though there seem to be some small inconsistencies in between the typing and the actual values
image
here trendingCollections is an array but the typing is for a singular item, it also references tokenPaginationInfo but that doesn't seem to be there anymore (as per John's comment, which makes sense)
image

File: packages/libs/api/sdk/src/graphql/fragmentMatcher.ts [hunk 4]
```diff
@@ -1,13 +1,26 @@
...
-    [key: string]: string[];
-  };
-}
-const result: PossibleTypesResultData = {
-  possibleTypes: {
-    Contract: ['BaseContract', 'ERC721Contract'],
-    Log: ['BaseLog', 'MintLog', 'OrderLog', 'TransferLog'],
-    Token: ['BaseToken', 'ERC721Token'],
-  },
+
```
[the remark this answers is not in the document]
Comment (AlejandroFabianCampos) [reply packages/libs/api/sdk/src/graphql/fragmentMatcher.ts:1]: gotcha, wasn't sure if this was autogen or manual to adapt, then nvm

**Review by johnpmitsch — APPROVED** [review]
All good from my perspective, I built the package and everything works as expected 👍🏻

**Review by AlejandroFabianCampos — APPROVED** [review]
LGTM, great work! 🚢

**Pull request merged**